### PR TITLE
DH-16

### PR DESCRIPTION
**Pull request by Dyrk84**
championAccuracy() and monsterAccuracy() reworked in Fight class.
Implemented methods in champion and monster sides: critical miss, critical miss events, deadly hit, critical hit.
Reworked damage() methods in class of the characters.
Fight class methods reordered in running time.

### CODE REVIEW

File: src/main/java/hu/dungeonhunter/characters/monsters/GoblinKing.java [hunk 1]
```diff
@@ -65,8 +72,9 @@ public boolean isDefeat() {
     }
 
     @Override
-    public int getMonsterDamage() {
-        return Dice.rollDice(maxDamage, numOfDices);
+    public int Damage() {
```
Comment (gergelyszanto) [inline src/main/java/hu/dungeonhunter/characters/monsters/GoblinKing.java]: Method names (with the exception of constructors) always start with lowercase.
This should be called: setDamage

File: src/test/java/hu/dungeonhunter/TestFight.java [hunk 2]
```diff
@@ -169,397 +179,302 @@ public void monsterCallerGoblinTest() {
...
-        Champion champion = new Champion();
-        champion.setHp(1 + fight.getMonster().getMaxDamage());
-        champion.setAccuracy(1000);
-        champion.setDefense(1000);
-        fight.setChampion(champion);
+        fight.setRandomCriticalMissEvent(5);
+        fight.attack(monster, 1);
 
-        fight.battle();
+        softly.assertThat(fight.getRandomCriticalMissEvent()).isLessThan(5);
```
Comment (gergelyszanto) [inline src/test/java/hu/dungeonhunter/TestFight.java:230]: Don't use soft assertion if there is only a single assertion in the test.

File: src/test/java/hu/dungeonhunter/TestFight.java [hunk 3]
```diff
@@ -169,397 +179,302 @@ public void monsterCallerGoblinTest() {
...
+        fight.attack(champion, 1);
 
-        ifAssertThatIsChampionStillAlive();
-        ifAssertThatIsMonsterStillAlive();
-        softly.assertThat(fight.getMonsterFactory().getKilledMonsterCounter()).isEqualTo(startingKilledMonsterCounter);
-        softly.assertThat(fight.getMonsterCounter()).isEqualTo(startingMonsterCounter);
-        softly.assertThat(fight.getChampion().getHealingPotionCounter()).isEqualTo(startingHealingPotionCounter);
-        softly.assertThat(fight.nextTurn()).isTrue();
+        softly.assertThat(fight.getRandomCriticalMissEvent()).isLessThan(5);
         softly.assertAll();
```
Comment (gergelyszanto) [inline src/test/java/hu/dungeonhunter/TestFight.java]: Same here, no need for softly asserting.

File: src/test/java/hu/dungeonhunter/TestFight.java [hunk 4]
```diff
@@ -169,397 +179,302 @@ public void monsterCallerGoblinTest() {
...
-        fight.setChampionFinalInitiation(1000);
-        MonstersInterface monster = monsterFactory.getMonster(CharacterTypes.GOBLIN);
-        monster.setDefense(1000);
+    public void monsterCriticalMissEvent1Test() {
+        Goblin monster = new Goblin();
+        monster.setHp(1);
         fight.setMonster(monster);
+        fight.setRandomCriticalMissEvent(1);
+        int[] monsterHpAfterDamage = {fight.getMonster().getHp() - fight.getMonster().getMaxDamage(),
+                fight.getMonster().getHp() - fight.getMonster().getNumOfDices()};
```
Comment (gergelyszanto) [inline src/test/java/hu/dungeonhunter/TestFight.java]: I believe this is incorrect, as you don't count in the number of dices to the may damage.
```suggestion
        int[] monsterHpAfterDamage = {fight.getMonster().getHp() - fight.getMonster().getMaxDamage() * fight.getMonster().getNumOfDices(),
                fight.getMonster().getHp() - fight.getMonster().getNumOfDices()};
```

File: src/test/java/hu/dungeonhunter/TestFight.java (hunk 2 above)
Comment (Dyrk84) [reply src/test/java/hu/dungeonhunter/TestFight.java:230]: fixed

File: src/test/java/hu/dungeonhunter/TestFight.java (hunk 3 above)
Comment (Dyrk84) [reply src/test/java/hu/dungeonhunter/TestFight.java]: Thank you, fixed!

File: src/test/java/hu/dungeonhunter/TestFight.java (hunk 4 above)
Comment (Dyrk84) [reply src/test/java/hu/dungeonhunter/TestFight.java]: Its true! I fixed this fail for all these types of  tests.

**Review by gergelyszanto — APPROVED** [review]

**Pull request merged**